From e7787ddb781a69338c5605ff3b57786bc4414bc5 Mon Sep 17 00:00:00 2001
From: Vivek Das Mohapatra <vivek@collabora.com>
Date: Wed, 1 Oct 2025 18:22:54 +0000
Subject: [PATCH 1/2] aarch64: fix -march=native aarch64_arches leading 'V'
 mismatch

In change 00c22ba69d8e738a4789b30165ff9c925c508fc1 a leading 'V'
was added to the aarch64-arches.def names (so that the name
could be used as part of valid C identifiers).

But it was not stripped from the call to get_arch_from_id
in host_detect_local_cpu in the -march path, which broke.
---
 gcc/config/aarch64/driver-aarch64.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/aarch64/driver-aarch64.cc b/gcc/config/aarch64/driver-aarch64.cc
index 9d99554dbc28..875aed6f0bdd 100644
--- a/gcc/config/aarch64/driver-aarch64.cc
+++ b/gcc/config/aarch64/driver-aarch64.cc
@@ -420,7 +420,7 @@ host_detect_local_cpu (int argc, const char **argv)
       if (arch)
 	{
 	  const char *arch_id = (aarch64_cpu_data[i].name
-				 ? aarch64_cpu_data[i].arch
+				 ? aarch64_cpu_data[i].arch + 1
 				 : DEFAULT_ARCH);
 	  auto arch_info = get_arch_from_id (arch_id);
 
-- 
2.39.5

